nautobot/extras/datasources/git.py

```diff
@@ -554,6 +571,155 @@ def delete_git_config_contexts(repository_record, job_result, preserve=(), prese
+    if not os.path.isdir(config_context_schema_path):
+        return
+
+    managed_config_context_schemas = set()
+
+    for file_name in os.listdir(config_context_schema_path):
+        if not os.path.isfile(os.path.join(config_context_schema_path, file_name)):
+            continue
+        job_result.log(
+            f"Loading config context schema from `{file_name}`",
+            grouping="config contexts",
+            logger=logger,
+        )
+        try:
+            with open(os.path.join(config_context_schema_path, file_name), "r") as fd:
+                # The data file can be either JSON or YAML; since YAML is a superset of JSON, we can load it regardless
+                try:
+                    context_schema_data = yaml.safe_load(fd)
+                except Exception as exc:
+                    raise RuntimeError(f"Error in loading config context schema data from `{file_name}`: {exc}")
+
+            # A file can contain one config context dict or a list thereof
+            if isinstance(context_schema_data, dict):
+                context_name = import_config_context_schema(context_schema_data, repository_record, job_result, logger)
+                managed_config_context_schemas.add(context_name)
+            elif isinstance(context_schema_data, list):
+                for context_schema in context_schema_data:
+                    context_name = import_config_context_schema(context_schema, repository_record, job_result, logger)
+                    managed_config_context_schemas.add(context_name)
+            else:
+                raise RuntimeError(
+                    f"Error in loading config context schema data from `{file_name}`: data must be a dict or list of dicts"
+                )
+        except Exception as exc:
+            job_result.log(
+                str(exc),
+                level_choice=LogLevelChoices.LOG_FAILURE,
+                grouping="config context schemas",
+                logger=logger,
+            )
+            job_result.save()
+
+    # Delete any prior contexts that are owned by this repository but were not created/updated above
+    delete_git_config_context_schemas(
+        repository_record,
+        job_result,
+        preserve=managed_config_context_schemas,
+    )
+
+
+def import_config_context_schema(context_schema_data, repository_record, job_result, logger):
+    """Using data from schema file, create schema record in Nautobot."""
+    git_repository_content_type = ContentType.objects.get_for_model(GitRepository)
+
+    created = False
+    modified = False
+
+    schema_metadata = context_schema_data.setdefault("_metadata", {})
+
+    if not schema_metadata.get("name"):
+        raise RuntimeError("File has no name set.")
+
+    try:
+        schema_record = ConfigContextSchema.objects.get(
+            name=schema_metadata["name"],
+            owner_content_type=git_repository_content_type,
+            owner_object_id=repository_record.pk,
```

Comment:
To match the pattern in the `ConfigContext` and `ExportTemplate` models, can you update in `nautobot/extras/models/models.py` the `ConfigContextSchema` model's `__str__` method so that it prepends the owner name if any?

```python
    def __str__(self):
        if self.owner:
            return f"[{self.owner}] {self.name}"
        return self.name
```